cadc-util/src/main/java/ca/nrc/cadc/auth/DelegationToken.java

```diff
@@ -140,27 +194,51 @@ public static String format(DelegationToken token)
     private static StringBuilder getContent(DelegationToken token)
     {
         StringBuilder sb = new StringBuilder();
-        sb.append("userid");
-        sb.append(VALUE_DELIM);
-        sb.append(token.getUser().getName());
-        if (StringUtil.hasText(token.getUser().getProxyUser()))
+
+        sb.append(EXPIRY_LABEL + VALUE_DELIM);
+        sb.append(token.getExpiryTime().getTime());
+
+        // Add all available identity principals to the content
+        // guarantee order
+        for (Principal prin: token.identityPrincipals) {
+            String tmpKey = "";
+            if (principalLabels.containsKey(prin.getClass().getSimpleName())) {
+                tmpKey = principalLabels.get(prin.getClass().getSimpleName());
+            } else {
+                continue;
```

Comment:
This "continue" is skipping an unrecognised IdentityType? That should at least be logged (at warn level) or fail. We should discuss impact of that choice.